File: GSDRequirementsCSharp.Web/scripts/App/04-Models/ClassDiagram/ClassProperty.js
var Models;
(function (Models) {
    var ClassProperty = (function () {
        function ClassProperty(data) {
            if (data === void 0) { data = null; }
            if (data) {
                for (var prop in data) {
                    this[prop] = data[prop];
                }
            }
        }
        ClassProperty.prototype.getDescription = function () {
            var visibility = Models.GetVisibilityUmlRepresentation(this.visibility);
            return visibility + " " + this.name + " : " + this.type;
        };
        return ClassProperty;
    })();
    Models.ClassProperty = ClassProperty;
})(Models || (Models = {}));

File: GSDRequirementsCSharp.Domain/Models/ClassDiagrams/ClassProperty.cs
namespace GSDRequirementsCSharp.Domain
{
    using Infrastructure.Persistence;
    using Models;
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    public class ClassProperty : IEntity<Guid>
    {
        public Guid Id { get; set; }

        public Guid ClassId { get; set; }

        public Visibility Visibility { get; set; }

        [Required]
        [StringLength(100)]
        public string Type { get; set; }

        public virtual Class Class { get; set; }

        public string Name { get; set; }
    }
}
